src/main/java/net/minecraftforge/event/terraingen/WorldTypeEvent.java

```diff
@@ -19,31 +19,43 @@
 
 package net.minecraftforge.event.terraingen;
 
+import javax.annotation.Nullable;
+import net.minecraft.world.World;
+import net.minecraft.world.WorldType;
 import net.minecraft.world.biome.BiomeProvider;
+import net.minecraft.world.gen.layer.GenLayer;
 import net.minecraftforge.common.MinecraftForge;
 import net.minecraftforge.fml.common.eventhandler.Cancelable;
 import net.minecraftforge.fml.common.eventhandler.Event;
-import net.minecraft.world.gen.layer.GenLayer;
-import net.minecraft.world.WorldType;
 
 /**
  * WorldTypeEvent is fired when an event involving the world occurs.<br>
  * If a method utilizes this {@link Event} as its parameter, the method will
  * receive every child event of this class.<br>
  * <br>
+ * {@link #world} the world for which the event is being fired. Can be null in some cases, for example when converting an old map to Anvil format. <br>
  * {@link #worldType} contains the WorldType of the world this event is occurring in.<br>
  * <br>
  * All children of this event are fired on the {@link MinecraftForge#TERRAIN_GEN_BUS}.<br>
  **/
 public class WorldTypeEvent extends Event
 {
+    @Nullable
+    private final World world;
     private final WorldType worldType;
 
-    public WorldTypeEvent(WorldType worldType)
+    public WorldTypeEvent(@Nullable World world, WorldType worldType)
```

Comment:
Public methods like this can't be changed directly, they need to be deprecated and replaced with new methods.